train.py

```diff
@@ -23,21 +23,36 @@
 from data import create_dataset
 from models import create_model
 from util.visualizer import Visualizer
+import torch.multiprocessing as mp
+import os
+import torch.distributed as dist
 
+def setup(rank, world_size,port):
+    os.environ['MASTER_ADDR'] = 'localhost'
+    os.environ['MASTER_PORT'] = port
 
-if __name__ == '__main__':
-    opt = TrainOptions().parse()   # get training options
-    dataset = create_dataset(opt)  # create a dataset given opt.dataset_mode and other options
+    # initialize the process group
+    dist.init_process_group("gloo", rank=rank, world_size=world_size)
```

Comment:
Use nccl instead of gloo (or option to choose backend?)

Reply:
Actually I think ability to choose backend is a good thing, so we may want an option for this